itest/psbt_test.go

```diff
@@ -1610,6 +1618,352 @@ func testPsbtSighashNoneInvalid(t *harnessTest) {
+
+	require.Len(t.t, bobVPsbt.Outputs, 1)
+	require.Equal(
+		t.t, bobVPsbt.Outputs[0].ScriptKey,
+		bobVPsbt.Outputs[0].Asset.ScriptKey,
+	)
+
+	// Bob restores Alice's signature for the asset input.
+	bobVPsbt.Outputs[0].Asset.PrevWitnesses = witnessBackup
+
+	bobVPsbtBytes, err := tappsbt.Encode(bobVPsbt)
+	require.NoError(t.t, err)
+
+	// Now let's serialize the edited vPSBT and commit it to our bitcoin
+	// PSBT.
+	b.Reset()
+	err = btcpsbt.Serialize(&b)
+	require.NoError(t.t, err)
+
+	// This call will also fund the PSBT, which means that the bitcoin that
+	// Alice "requested" previously by bumping her output will now be
+	// provided by Bob.
+	resp, err = bob.CommitVirtualPsbts(
+		ctxb, &wrpc.CommitVirtualPsbtsRequest{
+			VirtualPsbts: [][]byte{bobVPsbtBytes},
+			AnchorPsbt:   b.Bytes(),
+			AnchorChangeOutput: &wrpc.CommitVirtualPsbtsRequest_Add{
+				Add: true,
+			},
+			Fees: &wrpc.CommitVirtualPsbtsRequest_TargetConf{
+				TargetConf: 12,
+			},
+		},
+	)
+	require.NoError(t.t, err)
+
+	bobVPsbt, err = tappsbt.Decode(resp.VirtualPsbts[0])
+	require.NoError(t.t, err)
+
+	// Since Bob brings in a new input to the bitcoin transaction, he needs
+	// to sign it. We do not care about the sighash flag here, that can be
+	// the default, as we're not further editing the bitcoin transaction.
+	signResp := t.lndHarness.Bob.RPC.SignPsbt(
+		&walletrpc.SignPsbtRequest{
+			FundedPsbt: resp.AnchorPsbt,
+		},
+	)
+	require.NoError(t.t, err)
+
+	finalPsbt, err := psbt.NewFromRawBytes(
+		bytes.NewReader(signResp.SignedPsbt), false,
+	)
+	require.NoError(t.t, err)
+
+	// Bob must have brought his own input in order to pay Alice.
+	require.Len(t.t, finalPsbt.Inputs, 2)
+
+	// Bob's input should be at index 1, as index 0 is Alice's assets
+	// anchor.
+	bobInputIdx := uint32(1)
+
+	// Bob should sign exactly 1 input.
+	require.Len(t.t, signResp.SignedInputs, 1)
+
+	// Bob should have signed the input at the expected index.
+	require.Equal(t.t, bobInputIdx, signResp.SignedInputs[0])
+
+	require.NoError(t.t, finalPsbt.SanityCheck())
+
+	signedPkt := finalizePacket(t.t, t.lndHarness.Bob, finalPsbt)
+
+	require.True(t.t, signedPkt.IsComplete())
+
+	logResp := logAndPublish(
+		t.t, alice, signedPkt, []*tappsbt.VPacket{bobVPsbt}, nil, resp,
+	)
+
+	t.Logf("Logged transaction: %v", toJSON(t.t, logResp))
+
+	// Mine a block to confirm the transfer.
+	MineBlocks(t.t, t.lndHarness.Miner.Client, 1, 1)
+
+	// We also need to push the proof for this transfer to the universe
+	// server.
+	bobScriptKeyBytes := bobScriptKey.PubKey.SerializeCompressed()
+	sendUniProof(
+		t, t.universeServer.service, bob, bobScriptKeyBytes, genInfo,
+		mintedAsset.AssetGroup,
+		logResp.Transfer.Outputs[0].Anchor.Outpoint,
+	)
+
+	bobAssets, err := bob.ListAssets(ctxb, &taprpc.ListAssetRequest{})
+	require.NoError(t.t, err)
+
+	// Verify that Bob now holds the asset.
+	require.Len(t.t, bobAssets.Assets, 1)
+	require.Equal(t.t, bobAssets.Assets[0].Amount, numUnits)
+}
```

Comment:
We've got assertion functions to ensure that the transfer was complete. Maybe consider using https://github.com/lightninglabs/taproot-assets/blob/98634ca6d552d8eec4bb16491dc288f869f529f4/itest/assertions.go#L876-L876

Reply:
Tried using this but didn't seem to work? Will keep current checks for now